src/test/java/io/druid/data/input/MapBasedRowTest.java

```diff
@@ -51,4 +56,24 @@ public void testGetLongMetricFromString()
     Assert.assertEquals(-9223372036854775807L, row.getLongMetric("k5"));
     Assert.assertEquals(9223372036854775802L, row.getLongMetric("k6"));
   }
+
+  @Test
+  public void testMapWithNull()
+  {
+    Map<String, Object> map = new HashMap<>();
+    map.put("k0", null);
+    map.put("k1", Collections.singletonList(null));
+    map.put("k2", Arrays.asList(null, null, null));
+    MapBasedRow row = new MapBasedRow(new DateTime(), map);
+
+    // A) if value is NULL, then an empty list is returned
+    // (actually, its if (!map.containsKey()), with java8
+    // one should use getOrDefault)
+    Assert.assertEquals(Collections.emptyList(), row.getDimension("k0"));
+
+    // B) if value is a list, then _any_ NULL value is transformed to string "null"
```

Comment:
@okrische I don't think this is the behavior we want to see, null values should be maintained as is. There is no way for you to know, so it's not your fault. To give you some background information, Eric had outlined the behavior for `null` values in https://github.com/druid-io/druid/pull/995, so we should allow for null values within lists and not convert them to strings, otherwise this might cause some weird unexpected behavior.